ddpui/services/kpi_service.py

```
@@ -0,0 +1,534 @@
+                    if (
+                        len(trend) >= 2
+                        and trend[-1]["value"] is not None
+                        and trend[-2]["value"] is not None
+                        and trend[-2]["value"] != 0
+                    ):
+                        pop_change = round(
+                            ((trend[-1]["value"] - trend[-2]["value"]) / abs(trend[-2]["value"]))
+                            * 100,
+                            1,
+                        )
+                except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)